File: leetcode/0053.go
package leetcode

func maxSubArray(nums []int) int {
	mx, ans := 0, nums[0]
	for _, num := range nums {
		if mx+num > num {
			mx += num
		} else {
			mx = num
		}
		if ans < mx {
			ans = mx
		}
	}
	return ans
}

File: leetcode/0053.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int mx = 0, ans = nums[0];
        for (auto& i : nums) {
            mx = max(mx + i, i);
            if (mx > ans) {
                ans = mx;
            }
        }
        return ans;
    }
};